src/nuis/measurement/IRecord.h

```diff
@@ -19,67 +19,44 @@
 #pragma once
 #include <vector>
 #include <string>
+#include <memory>
 
 #include "yaml-cpp/yaml.h"
 #include "HepMC3/GenEvent.h"
 #include "ProSelecta/ProSelecta.h"
 
-#include "nuis/measurement/Record.h"
+#include "nuis/measurement/Projection.h"
 #include "nuis/measurement/Variables.h"
 #include "nuis/measurement/Document.h"
 
 namespace nuis {
 namespace measurement {
 
-class MeasurementLoader {
-public:
+using ProjectionPtr = std::shared_ptr<nuis::measurement::Projection>;
 
-  MeasurementLoader() {
-  }
+class IRecord {
+ public:
+  IRecord() {}
 
-  explicit MeasurementLoader(YAML::Node config) {
-  }
-
-  inline virtual ~MeasurementLoader() {
-  }
+  inline virtual ~IRecord() {}
 
   inline virtual std::vector<double>
-    ProjectEvent(const HepMC3::GenEvent& event) {
-    return std::vector<double>();
-  }
-
-  inline virtual bool
-    FilterEvent(const HepMC3::GenEvent& event) {
-    return false;
-  }
-
-  inline virtual double
-    WeightEvent(const HepMC3::GenEvent& event) {
-      return 1.0;
-  }
-
-  virtual Record CreateRecord(const std::string label = "MC") {
-      return Record();
-  }
+    ProjectEvent(const HepMC3::GenEvent& event) = 0;
 
-  virtual bool FillRecordFromEvent(
-    Record& h, const HepMC3::GenEvent& event, const double weight) {
-  }
+  inline virtual bool FilterEvent(const HepMC3::GenEvent& event) = 0;
 
-  virtual bool FillRecordFromProj(
-    Record& h, const std::vector<double>& val, const double weight) {
-  }
+  inline virtual double WeightEvent(const HepMC3::GenEvent& event) = 0;
```

Comment:
God I'm being picky, but this one doesn't actually weight the event, it would be better to call it EventWeight or GetEventWeight or  something.